src/multiplex/client.rs

```diff
@@ -77,32 +143,40 @@ pub enum SpawnError<E> {
     Inner(E),
 }
 
-impl<NT, Target, Request> Service<Target> for Maker<NT, Request>
+impl<NT, P, Target, Request> Service<Target> for Maker<NT, P, Request>
 where
     NT: MakeTransport<Target, Request>,
     NT::Transport: 'static + Send + TagStore<Request, NT::Item>,
     <NT::Transport as TagStore<Request, NT::Item>>::Tag: 'static + Send,
+    P: PendingStore<NT::Transport, Request> + Default + Send + 'static,
```

Comment:
Yeah the issue is that MakeTransport only makes a single item and I wanted to avoid that complexity. Though you're right its not ideal. I think `Default` can work since usually you don't need any input parameters or can customize those by doing a wrapping newtype default impl.

Reply:
I'm thinking of things where you want to provide the `PendingStore` with non-global state, like a special allocator or something along those lines. `Default` basically precludes that without significant hacks.

Reply:
Right, I agree that does sound like a niche case but I am not actually sure how we could support that via the MakeTransport trait? Do you have any suggestions?